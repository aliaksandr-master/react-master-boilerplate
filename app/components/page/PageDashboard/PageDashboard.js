import { compose, setDisplayName } from 'recompose';
import { connect } from 'react-redux-self';
import reducer from './PageDashboard.store';
import PageDashboardView from './PageDashboard.view';



export default compose(
  setDisplayName('PageDashboard'),
  connect({
    reducer,
    getters: [
      (ownStoreState) => ownStoreState
    ],
    selector: (ownStoreState) => ({
      ...ownStoreState
    }),
    denormalize: null,
    mapDispatchToProps: null
  }),
)(PageDashboardView);
