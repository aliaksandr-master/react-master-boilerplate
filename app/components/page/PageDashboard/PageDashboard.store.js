import { Dux } from '../../../lib/reducer';



const dux = Dux('PageDashboard');
//const DO_SOMETHING = dux.actionType('LOAD_SOMETHING');


//export const loadSomethingActionCreator = dux.actionCreator(DO_SOMETHING, (payload, arg2, arg3) => payload);


dux.reducer({
}, {
  //  [DO_SOMETHING]: (state, { payload }) => ({
  //    ...state
  //  })
  //})
});

export default dux.compileReducer();
