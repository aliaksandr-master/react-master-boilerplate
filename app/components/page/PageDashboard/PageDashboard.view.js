import React from 'react';
import PropTypes from 'prop-types';
import classnames from '../../../vendor/classnames';
import './PageDashboard.view.less';



const PageDashboard = ({ className }) => {
  return (
    <div className={classnames('PageDashboard', className)}>
      Hello!
    </div>
  );
};



PageDashboard.propTypes = {
  className: PropTypes.string
};



PageDashboard.defaultProps = {
};



export default PageDashboard;
