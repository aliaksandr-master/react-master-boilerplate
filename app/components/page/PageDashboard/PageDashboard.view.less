@import '../../../lib/mixins.less';

.PageDashboard {
}

