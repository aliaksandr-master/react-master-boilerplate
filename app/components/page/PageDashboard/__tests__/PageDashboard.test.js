/* eslint-env browser, jest */

import React from 'react';
import renderer from 'react-test-renderer';
import PageDashboard from '../index';
import Test from '../../../test/Test';



describe('rendering', () => {
  it('should render something', () => {
    const component = renderer.create(
      <Test>
        {() => <PageDashboard />}
      </Test>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
