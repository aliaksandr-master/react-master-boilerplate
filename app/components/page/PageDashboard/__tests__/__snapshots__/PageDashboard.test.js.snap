// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render something 1`] = `
<div
  className="PageDashboard"
>
  Hello!
</div>
`;
